api/app/channels/registry.py

```
@@ -0,0 +1,430 @@
+
+            try:
+                await asyncio.wait_for(entry.plugin.stop(), timeout=timeout)
+                entry.started = False
+
+                # Call on_shutdown hook if available
+                if hasattr(entry.plugin, "on_shutdown"):
+                    await entry.plugin.on_shutdown()
+
+                logger.info(f"Stopped channel '{channel_id}'")
+
+            except Exception as e:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)